python/pylibcugraph/pylibcugraph/node2vec.pyx

```diff
@@ -0,0 +1,184 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Have cython use python 3 syntax
+# cython: language_level = 3
+
+from libc.stdint cimport uintptr_t
+
+from pylibcugraph._cugraph_c.cugraph_api cimport (
+    bool_t,
+    data_type_id_t,
+    cugraph_resource_handle_t,
+)
+from pylibcugraph._cugraph_c.error cimport (
+    cugraph_error_code_t,
+    cugraph_error_t,
+)
+from pylibcugraph._cugraph_c.array cimport (
+    cugraph_type_erased_device_array_view_t,
+    cugraph_type_erased_device_array_view_create,
+    cugraph_type_erased_device_array_free,
+)
+from pylibcugraph._cugraph_c.graph cimport (
+    cugraph_graph_t,
+)
+from pylibcugraph._cugraph_c.algorithms cimport (
+    cugraph_node2vec,
+    cugraph_random_walk_result_t,
+    cugraph_random_walk_result_get_paths,
+    cugraph_random_walk_result_get_weights,
+    cugraph_random_walk_result_get_path_sizes,
+    cugraph_random_walk_result_free,
+)
+from pylibcugraph.resource_handle cimport (
+    EXPERIMENTAL__ResourceHandle,
+)
+from pylibcugraph.graphs cimport (
+    _GPUGraph,
+)
+from pylibcugraph.utils cimport (
+    assert_success,
+    copy_to_cupy_array,
+    assert_CAI_type,
+    get_c_type_from_numpy_type,
+)
+
+
+def EXPERIMENTAL__node2vec(EXPERIMENTAL__ResourceHandle resource_handle,
+                           _GPUGraph graph,
+                           src_array,
```

Comment:
Probably rename it `seed _array` to avoid confusion. In fact `src_array` already has a meaning internally in `EXPERIMENTAL__SGGraph(EXPERIMENTAL__Graph)`